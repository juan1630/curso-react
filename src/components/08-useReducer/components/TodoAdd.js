import React from 'react';
import { useForm } from '../../../hooks/useHook'

export const TodoAdd = ({handleAddTodo}) => {

    const [ { description}, handleInputChange, reset  ] =  useForm({
        
        description:''
   
       });


       const handleOnSubmit = (e) =>{ 
            
        e.preventDefault();
        if( description.trim().length < 1 ){
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        }


        handleAddTodo( newTodo)
        // reseteamos el form 
        reset();

       
    }
   
    return (
        <>
             <h3 className="agregar" >
                        Agregar todo
                   </h3>


                   <hr/>

                    <form onSubmit={  handleOnSubmit }  >


                        <input 
                        type="text"
                        className="form-control"
                        name="description"
                        placeholder="Comer..."
                        autoComplete="off"
                        onChange={ handleInputChange }
                        value={ description }
                        />


                        <button className="btn btn-success btn-block" > Agregar </button>

                    </form>
        </>
    )
}
