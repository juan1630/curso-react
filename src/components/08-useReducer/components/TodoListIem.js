import React from 'react'


export const TodoListIem = ({ todo, i, handleDelete, handleToggle }) => {
    // console.log(todo, i, handleDelete, handleToggle )
    return (
        <div>
             <li  onClick={ ()=> { handleToggle(todo.id) } } className={ `${todo.done && 'complete'} ` }  key={todo.id} >  { i +1 } - {todo.desc} 
                            <hr/>
                            <button 
                            
                            className="btn btn-danger"
                            onClick={ () => { handleDelete( todo.id ) } }
                            > 

                                Borrar
                            </button>
                        
                          </li>
        </div>
    )
}
