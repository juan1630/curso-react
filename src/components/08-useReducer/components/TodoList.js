import React from 'react';
import { TodoListIem } from './TodoListIem'

export const TodoList = ( { todos = [] , handleDelete, handleToggle} ) => {

    console.log( todos );
    
    return (

               <ul className="todos-items" >
                {
                    todos.map( (todo, i) => (
                        
                        <TodoListIem 
                        key={ todo.id }
                        todo={ todo} 
                         i={ i }   
                        handleDelete ={handleDelete} 
                        handleToggle={handleToggle} />
                       
                    ))
                }
            </ul>
    )
}
