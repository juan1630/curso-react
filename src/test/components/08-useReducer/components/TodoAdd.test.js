import React from 'react';
import {  shallow  } from 'enzyme';
import {  TodoAdd } from '../../../../components/08-useReducer/components/TodoAdd'

describe('Pruebas en el componente TODOADD', ()=> {

    const handleAddTodo = jest.fn();

    const wrapper = shallow( <TodoAdd 
            handleAddTodo={ handleAddTodo }

            /> )

    test('Debe de mostrarse correctamente', () => {

        expect(  wrapper).toMatchSnapshot();

    });

    test('No debe de llamr al TODO', ()=> {

        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({ preventDefault () {} });

        expect( handleAddTodo ).toHaveBeenCalledTimes(0);
        // esperamos que se haya llamado 0 veces la funcion

    });

    test('debe de llamar la funcion handleAddTodo', () => {
            
            const value = 'Aprender react';
            
            wrapper.find('input').simulate('change', {
                target: {
                    value,
                    name: 'description'
                }
                // etse name es del elemento html
            })

            const formSubmit = wrapper.find('form').prop('onSubmit');

            formSubmit({  preventDefault(){} });
            
            expect( handleAddTodo ).toHaveBeenCalledTimes( 1 );
            expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object ) );
            expect( handleAddTodo ).toHaveBeenCalledWith({
                desc: value,
                done: false,
                id: expect.any(Number)
            });

            expect( wrapper.find('input').prop('value') ).toBe('')
    })
    

});