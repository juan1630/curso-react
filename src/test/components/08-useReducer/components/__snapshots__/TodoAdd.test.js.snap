// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente TODOADD Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="agregar"
  >
    Agregar todo
  </h3>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Comer..."
      type="text"
      value=""
    />
    <button
      className="btn btn-success btn-block"
    >
       Agregar 
    </button>
  </form>
</Fragment>
`;
